src/plugin/storage/persistplugin.js

```diff
@@ -0,0 +1,131 @@
+goog.module('plugin.storage.PersistPlugin');
+goog.module.declareLegacyNamespace();
+
+goog.require('os.ui.window.confirmDirective');
+const browser = goog.require('goog.labs.userAgent.browser');
+const log = goog.require('goog.log');
+const AlertEventSeverity = goog.require('os.alert.AlertEventSeverity');
+const AbstractPlugin = goog.require('os.plugin.AbstractPlugin');
+const Metrics = goog.require('os.metrics.Metrics');
+const LOGGER_ = log.getLogger('PersistPlugin');
+
+const PERSISTENCE_SUCCEEDED = 'storage.persistence.succeeded';
+const PERSISTENCE_ACCEPTED = 'storage.persistence.accepted';
+const PERSISTENCE_DECLINED = 'storage.persistence.declined';
+const PERSISTENCE_FAILED = 'storage.persistence.failed';
+
+
+/**
+ * Plugin to check for persistent storage and ask the user
+ * to give permission for persistent storage.
+ */
+class PersistPlugin extends AbstractPlugin {
+  /**
+   */
+  constructor() {
+    super();
+    this.id = 'persist';
+  }
+
+  /**
+   * @inheritDoc
+   */
+  init() {
+    const msgs = [];
+    const metrics = Metrics.getInstance();
+
+    if (navigator.storage && navigator.storage.persist) {
+      navigator.storage.persisted().then((result) => {
+        if (!result) {
+          navigator.storage.persist().then((userDecision) => {
+            if (!userDecision) {
+              msgs.length = 0;
+              msgs.push('By declining persistent storage, your application settings may be automatically ' +
+                'reset when under storage pressure (the conditions for that event are browser dependent).');
+              this.log(msgs);
+              metrics.updateMetric(PERSISTENCE_DECLINED, 1);
+            } else {
+              log.info(LOGGER_, 'User allowed (or browser automatically allowed) persistent storage');
+              metrics.updateMetric(PERSISTENCE_ACCEPTED, 1);
+              metrics.updateMetric(PERSISTENCE_SUCCEEDED, 1);
+            }
+          });
+        } else {
+          log.info(LOGGER_, 'Storage is persistent');
+          metrics.updateMetric(PERSISTENCE_SUCCEEDED, 1);
+        }
+      });
+    } else {
+      if (window.isSecureContext === false) {
+        msgs.push('The application is not being served in a secure context. Persistent storage permission ' +
+            'will be automatically declined.');
+      } else if (browser.isChrome()) {
```

Comment:
I think this check needs to be in the `if (!userDecision)` block. In Chrome, `navigator.storage.persist()` will be called, but Chrome makes that decision without user input.